src/VisualStudio/Xaml/Impl/Implementation/XamlTextViewCreationListener_IVsSolutionEvents.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
+using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    internal partial class XamlTextViewCreationListener : IVsSolutionEvents
```

Comment:
Correct, I need to listen to this one in order to disconnect the XamlProject when the real project closes.